spark/src/main/scala/com/cloudera/livy/spark/client/ClientSession.scala

```diff
@@ -80,6 +94,18 @@ class ClientSession(val sessionId: Int, createRequest: CreateClientRequest)
     new JobStatus(id, status.state, status.result, status.error)
   }
 
+  private def copyResourceToHDFS(data: Array[Byte], name: String): URI = {
```

Comment:
So, this method has a few security implications... since we don't have security hooked up yet it's probably ok, but in the long time we need at least a big "TODO: " comment here to remind us.

Basically, with proper security in place, you can't write these files as the Livy user, because then any session could read the files, not just sessions belonging to the user uploading them.